lms/djangoapps/student_account/views.py

```diff
@@ -74,6 +74,30 @@ def login_and_registration_form(request, initial_mode="login"):
     if UserProfile.has_registered(request.user):
         return redirect(redirect_to)
 
+    if third_party_auth.is_enabled():
+        force_provider_id = settings.FEATURES.get('FORCE_TPA')
+        if force_provider_id:
+            force_provider = third_party_auth.provider.Registry.get(
+                provider_id=force_provider_id
+            )
+            if force_provider and force_provider.display_for_login:
+                running_pipeline = third_party_auth.pipeline.get(request)
+                if not running_pipeline:
+                    if initial_mode == "login":
```

Comment:
Since the values for `AUTH_ENTRY_LOGIN` and `AUTH_ENTRY_REGISTER` are `login` and `register`, which is to say, the same as the values of `initial_mode` that we're checking here, can we simplify this like:

```python
# Put this at the top-level of the file, as a constant.
REDIRECTABLE_MODES = [
    pipeline.AUTH_ENTRY_LOGIN,
    pipeline.AUTH_ENTRY_REGISTER,
]
# ...
# then down here...
if initial_mode in REDIRECTABLE_MODES:
    tpa_url = pipeline.get_login_url(
        force_provider_id,
        initial_mode,
        redirect_url=redirect_to,
    )
    return redirect(tpa_url)
```